File: problem-solving/algorithms/between-two-sets/between-two-sets.js
function getTotalX(a, b) {
    function calculateGcd(x, y) {
        while (y) {
            let temp = y;
            y = x % y;
            x = temp;
        }
        return x;
    }

    function calculateLcm(x, y) {
        return x * y / calculateGcd(x, y);
    }

    let lcmA = a[0];
    for (let i = 1; i < a.length; i++) {
        lcmA = calculateLcm(lcmA, a[i]);
    }

    let gcdB = b[0];
    for (let i = 1; i < b.length; i++) {
        gcdB = calculateGcd(gcdB, b[i]);
    }

    let count = 0;
    for (let i = lcmA; i <= gcdB; i += lcmA) {
        if (gcdB % i === 0) {
            count++;
        }
    }

    return count;
}

File: problem-solving/algorithms/between-two-sets/between_two_sets.py
def getTotalX(a, b):
    def calculate_gcd(x, y):
        while y:
            x, y = y, x % y
        return x

    def calculate_lcm(x, y):
        return x * y // calculate_gcd(x, y)

    lcm_a = a[0]
    for i in a[1:]:
        lcm_a = calculate_lcm(lcm_a, i)

    gcd_b = b[0]
    for i in b[1:]:
        gcd_b = calculate_gcd(gcd_b, i)

    count = 0
    for i in range(lcm_a, gcd_b + 1, lcm_a):
        if gcd_b % i == 0:
            count += 1

    return count